src/db.rs

```diff
@@ -38,30 +39,27 @@ impl DB {
         if self.modified {
             let path_tmp = self.get_path_tmp();
 
-            let file_tmp = File::create(&path_tmp)
-                .with_context(|| anyhow!("could not open temporary database file"))?;
+            let file_tmp =
+                File::create(&path_tmp).context("could not open temporary database file")?;
 
             let writer = BufWriter::new(&file_tmp);
-            bincode::serialize_into(writer, &self.dirs)
-                .with_context(|| anyhow!("could not serialize database"))?;
+            bincode::serialize_into(writer, &self.dirs).context("could not serialize database")?;
 
-            fs::rename(&path_tmp, &self.path)
-                .with_context(|| anyhow!("could not move temporary database file"))?;
+            fs::rename(&path_tmp, &self.path).context("could not move temporary database file")?;
         }
 
         Ok(())
     }
 
     pub fn migrate<P: AsRef<Path>>(&mut self, path: P, merge: bool) -> Result<()> {
         if !self.dirs.is_empty() && !merge {
-            bail!(
+            bail!(indoc!(
```

Comment:
Similar to the above: I see no reason to use a crate that pulls in 4 transitive dependencies, when `"{}\n{}"` or the previous solution works just fine. Maybe this is me trying to be minimalist, but this is used only once: in this single line...

As you can probably gather, I'm not a fan of `left-pad`-like dependencies, even though Rust makes them so easy to use.

Reply:
The only overhead of `indoc` is at compile time - it does not have any impact on the output binary.

I included this primarily because I was hoping to use it for https://github.com/ajeetdsouza/zoxide/issues/32. The issue there is that I will have to replace the large static strings by formatted strings, generated inside functions. I would really have liked indented strings there, because not having them really screws up code collapsing on my editor, which is frustrating (especially for large functions/strings).

The venture proved pointless, however - `indoc` only supports formatted strings on nightly Rust. For now, since there is no runtime impact, I'm thinking of leaving it in all the same (it might prove useful elsewhere), but I would gladly hear your take on the matter.

Reply:
I don't disagree that there is no runtime impact. However, there is a compile-time impact -- 4 more crates need to be compiled and linked. Sure, this isn't much of a problem now, but it might become one down the line. As implied in my original comment, dependencies that can be implemented solely using the standard library irk me.

If it does prove useful elsewhere, then that would be a perfect opportunity to add it! But keeping it around "just in case" seems like a waste to me.

Reply:
That's true, I will remove the dependency.